packages/dds/sequence/src/test/sharedString.spec.ts

```diff
@@ -236,6 +236,29 @@ describe("SharedString", () => {
 			assert.equal(simpleMarker.properties?.color, "blue", "Could not annotate marker");
 		});
 
+		it("fails when the marker is updated", () => {
+			sharedString.insertText(0, "hello world");
+			// Insert a simple marker.
+			sharedString.insertMarker(6, ReferenceType.Simple, {
+				[reservedMarkerIdKey]: "markerId",
+			});
+			// Annotate the marker.
+			const props = { color: "blue" };
+			const simpleMarker = sharedString.getMarkerFromId("markerId") as Marker;
+			sharedString.annotateMarker(simpleMarker, props);
+			assert.equal(simpleMarker.properties?.color, "blue", "Could not annotate marker");
+			// Annotate the marker's ID.
+			const newIdProps = { [reservedMarkerIdKey]: "newIdValue" };
+			try {
+				sharedString.annotateMarker(simpleMarker, newIdProps);
+			} catch (error) {
```

Comment:
It looks like this test can never fail. If `annotateMarker` marker doesn't throw, the `catch` block is never reached. There are a couple ways to solve this; notably you can use `assert.fail` after the `annotateMarker` call to force an exception (note that this would be caught by the `catch`).